src/callbacks/dump_model_graph.cpp

```diff
@@ -0,0 +1,155 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+#include <boost/graph/graphviz.hpp>
```

Comment:
I am concerned that this is starting to pull in the boost library.  Is there another alternative that we can use?  I am afraid that this will start to greatly complicate the build environment.

Reply:
@bvanessen This callback only requires Boost's header flies to build, and the LC environments (at least Lassen, Pascal, and Ray) has Boost in `/usr/include`, so this callback doesn't have to change the current build configurations.
For other machines, I added a new commit that disables the callback when Boost is unavailable.